File: golang/1_two_sum.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
	"strings"
)

func twoSum(nums []int, target int) []int {
	var emptyArr []int
	m := make(map[int]int)
	for i := 0; i < len(nums); i++ {
		diff := target - nums[i]
		if index, ok := m[diff]; ok {
			res := []int{index, i}
			return res
		} else {
			m[nums[i]] = i
		}
	}
	return emptyArr
}

func main() {
	var nums []int
	var target int
	scanner := bufio.NewScanner(os.Stdin)
	scanner.Scan()
	line := scanner.Text()
	splittedInput := strings.Split(line, ", ")
	for i := range splittedInput {
		num, err := strconv.Atoi(splittedInput[i])
		if err == nil {
			nums = append(nums, num)
		}
	}
	scanner.Scan()
	target, err := strconv.Atoi(scanner.Text())
	if err != nil {
		target = -1
	}
	fmt.Println(twoSum(nums, target))
}

File: python/1_two_sum.py
from typing import List

class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        temp_num = dict()
        for i in range(0, len(nums)):
            need = target - nums[i]
            if need in temp_num:
                return [temp_num[need], i]
            else:
                temp_num[nums[i]] = i
        return []

sol = Solution()
nums = [int(x) for x in input().split(', ')]
target = int(input())
print(sol.twoSum(nums, target))